arangosh/Import/quick_hist.h

```diff
@@ -0,0 +1,265 @@
+    interval_diff=std::chrono::duration_cast<std::chrono::milliseconds>(interval_end - _interval_start);
+
+    {
+      // retest within mutex
+      interval_end=std::chrono::steady_clock::now();
+      interval_diff=std::chrono::duration_cast<std::chrono::milliseconds>(interval_end - _interval_start);
+      if (std::chrono::milliseconds(10000) <= interval_diff || force) {
+        double fp_measuring, fp_interval;
+        size_t num=_readingLatencies->size();
+
+        measuring_diff = std::chrono::duration_cast<std::chrono::milliseconds>(interval_end - _measuring_start);
+        fp_measuring = measuring_diff.count() / 1000.0;
+        fp_interval = interval_diff.count() / 1000.0;
+
+        std::chrono::microseconds mean, median, per95, per99, per99_9;
+
+        if (0 != num) {
+          bool odd(num & 1);
+          size_t half(num/2);
+
+          sum = std::accumulate(_readingLatencies->begin(), _readingLatencies->end(), zero_micros);
+          mean = sum / num;
+
+          if (1==num) {
+            median = _readingLatencies->at(0);
+          } else if (odd) {
+            median = (_readingLatencies->at(half) + _readingLatencies->at(half +1)) / 2;
+          } else {
+            median = _readingLatencies->at(half);
+          }
+        } else {
+          sum = zero_micros;
+          mean = zero_micros;
+          median = zero_micros;
+        } // else
+
+        sort(_readingLatencies->begin(), _readingLatencies->end());
+
+        // close but not exact math for percentiles
+        per95 = CalcPercentile(*_readingLatencies, 950);
+        per99 = CalcPercentile(*_readingLatencies, 990);
+        per99_9 = CalcPercentile(*_readingLatencies, 999);
+
+        // timestamp to help match to other logs ...
+        auto t = std::time(nullptr);
+        auto tm = *std::localtime(&t);
+
+        // new age string buffering & formatting
+        std::ostringstream oss;
+        oss << std::put_time(&tm, "%m-%d-%Y %H:%M:%S");
+        auto str = oss.str();
+
+        // old age string buffering & formatting
+        char buffer[133];
+        snprintf(buffer, sizeof(buffer), "%.3f,%.3f,%zd,%ld,%ld,%ld,%ld,%ld,%ld,%ld,%d,%s",
+               fp_measuring, fp_interval, num,
+               (0!=num) ? _readingLatencies->at(0).count() : 0,
+               mean.count(),median.count(),
+               per95.count(), per99.count(), per99_9.count(),
+               (0!=num) ? _readingLatencies->at(num-1).count() : 0,
+               0, str.c_str());
+        LOG_TOPIC(INFO, arangodb::Logger::FIXME)
+          << buffer;
+
+        _readingLatencies->clear();
+        _interval_start=interval_end;
+      }
+    }
+  }
+
+  //
+  // calculation taken from http://www.dummies.com/education/math/statistics/how-to-calculate-percentiles-in-statistics
+  //  (zero and one size vector calculations not included in that link)
+  std::chrono::microseconds CalcPercentile(std::vector<std::chrono::microseconds> &SortedLatencies, int Percentile) {
```

Comment:
According to our coding conventions, we put both the `&` and the `*` on the left-hand side.

Reply:
And we wouldn't use function parameter names starting with an upper case letter.

Reply:
i think fixed